tests/functional/aws-node-sdk/test/multipleBackend/objectPutCopyPart/objectPutCopyPartAzure.js

```diff
@@ -0,0 +1,488 @@
+        beforeEach(() => {
+            bucketUtil = new BucketUtility('default', sigCfg);
+            s3 = bucketUtil.s3;
+        });
+
+        afterEach(() => {
+            process.stdout.write('Emptying bucket\n');
+            return bucketUtil.empty(azureContainerName)
+            .then(() => {
+                process.stdout.write('Deleting bucket\n');
+                return bucketUtil.deleteOne(azureContainerName);
+            })
+            .catch(err => {
+                process.stdout.write(`Error in afterEach: ${err}\n`);
+                throw err;
+            });
+        });
+        describe('Basic test: ', () => {
+            beforeEach(function beF(done) {
+                this.currentTest.keyNameNormalAzure =
+                `normalazure${uniqName(keyObjectAzure)}`;
+                this.currentTest.keyNameNormalAzureMismatch =
+                `normalazuremismatch${uniqName(keyObjectAzure)}`;
+
+                this.currentTest.keyNameFiveMbAzure =
+                `fivembazure${uniqName(keyObjectAzure)}`;
+                this.currentTest.keyNameFiveMbMem =
+                `fivembmem${uniqName(keyObjectMemory)}`;
+
+                this.currentTest.mpuKeyNameAzure =
+                `mpukeyname${uniqName(keyObjectAzure)}`;
+                const params = {
+                    Bucket: azureContainerName,
+                    Key: this.currentTest.mpuKeyNameAzure,
+                    Metadata: { 'scal-location-constraint': azureLocation },
+                };
+                async.waterfall([
+                    next => s3.createBucket({ Bucket: azureContainerName },
+                      err => next(err)),
+                    next => s3.putObject({
+                        Bucket: azureContainerName,
+                        Key: this.currentTest.keyNameNormalAzure,
+                        Body: normalBody,
+                        Metadata: { 'scal-location-constraint': azureLocation },
+                    }, err => next(err)),
+                    next => s3.putObject({
+                        Bucket: azureContainerName,
+                        Key: this.currentTest.keyNameNormalAzureMismatch,
+                        Body: normalBody,
+                        Metadata: { 'scal-location-constraint':
+                        azureLocationMismatch },
+                    }, err => next(err)),
+                    next => s3.putObject({
+                        Bucket: azureContainerName,
+                        Key: this.currentTest.keyNameFiveMbAzure,
+                        Body: fiveMbBody,
+                        Metadata: { 'scal-location-constraint': azureLocation },
+                    }, err => next(err)),
+                    next => s3.putObject({
+                        Bucket: azureContainerName,
+                        Key: this.currentTest.keyNameFiveMbMem,
+                        Body: fiveMbBody,
+                        Metadata: { 'scal-location-constraint': memLocation },
```

Comment:
let's try not to call things memLocation.  Thinking about porting the tests to integration, it will probably make sense to have something like localLocation and in integration that could be sproxyd.